### PR TITLE
Stateless manager

### PR DESCRIPTION
**Pull request by rsgoncalves**

### CODE REVIEW

File: src/main/java/edu/stanford/protege/metaproject/impl/ConfigurationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,843 @@
...
+     * @return ServerConfigurationBuilder
+     */
+    public ConfigurationBuilder setProject(ProjectId projectId, Project project) {
+        checkNotNull(projectId);
+        checkNotNull(project);
+        getProject(projectId).ifPresent(this::removeProject);
+        try {
+            addProject(project);
+        } catch (IdAlreadyInUseException e) {
+            logger.debug("Attempted to replace a project but the original project instance was not properly deleted beforehand");
```
Comment (bdionne) [inline src/main/java/edu/stanford/protege/metaproject/impl/ConfigurationBuilder.java]: so if this is swallowed the caller will never see that no change occurs?

File: src/main/java/edu/stanford/protege/metaproject/impl/ConfigurationBuilder.java [hunk 2]
```diff
@@ -0,0 +1,843 @@
...
+     * @return ServerConfigurationBuilder
+     */
+    public ConfigurationBuilder setRole(RoleId roleId, Role role) {
+        checkNotNull(roleId);
+        checkNotNull(role);
+        getRole(roleId).ifPresent(this::removeRole);
+        try {
+            addRole(role);
+        } catch (IdAlreadyInUseException e) {
+            logger.debug("Attempted to replace a role but the original role instance was not properly deleted beforehand");
```
Comment (bdionne) [inline src/main/java/edu/stanford/protege/metaproject/impl/ConfigurationBuilder.java]: same as above

**Comment by bdionne** [comment]
runs fine with rest of system, LGTM

File: src/main/java/edu/stanford/protege/metaproject/impl/ConfigurationBuilder.java (hunk 1 above)
Comment (rsgoncalves) [inline src/main/java/edu/stanford/protege/metaproject/impl/ConfigurationBuilder.java]: These set methods replace existing policy elements: If the given projectId does not exist, the code neither removes nor adds anything. If the projectId exists, then the project with that id is deleted and replaced with the given one. The addition can only fail if deletion fails (the element with that id lives on), and delete can only fail if an element with that id doesn't exist.

I'll add some extra logging for stating when these identifiers don't exist.

Comment (bdionne) [inline src/main/java/edu/stanford/protege/metaproject/impl/ConfigurationBuilder.java]: hmm, confusing. If the delete fails (id doesn't exist), then the add will succeed, non? If the add can only fail if the delete fails, and the delete fails when the id does not exist, then the exception doesn't seem right.

But still isn't it the case that to the caller the set does not actually happen?

**Pull request merged**